extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -134,18 +134,18 @@ public String getValue(Widget w) {
     }
 
     /**
-     * Escapes the path part of a URL as defined in RFC2396. This means, that for example the
-     * path "/hello world" gets escaped to "/hello%20world".
+     * Escapes parts of a URL. This means, that for example the
+     * path "/hello world" gets escaped to "/hello+world".
      *
-     * @param path The path of the URL that has to be escaped
-     * @return The escaped path
+     * @param string The string that has to be escaped
+     * @return The escaped string
      */
-    protected String escapeURLPath(String path) {
+    protected String escapeURL(String string) {
         try {
-            return new URI(null, null, path, null).toString();
-        } catch (URISyntaxException use) {
-            logger.warn("Cannot escape path '{}' in URL. Returning unmodified path.", path);
-            return path;
+            return URLEncoder.encode(string, "UTF-8");
```

Comment:
Will this take `/path/to/resource` and turn it into `%2Fpath%2Fto%2Fresource` (according to [this](https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html) it will), and is that what you want?